### PR TITLE
chore: pull helio add test for tls deadlock

### PR DESCRIPTION
**Pull request by kostasrim**
* pull helio
* add test that covers tls deadlock

### CODE REVIEW

**Comment by kostasrim** [comment]
fails to build locally:

```
   17 |   static_assert(kSegLogSpan == kSegmentShift);
   18 |   static_assert((~kSegmentAlignMask) == (MI_SEGMENT_SIZE - 1));
```

Seems like minalloc changed SegmentSize

**Comment by romange** [comment]
Seems there are also issues with glog, though I am not sure why, are you investigating this?

**Comment by kostasrim** [comment]
> Seems there are also issues with glog, though I am not sure why, are you investigating this?

yes I will take care of this

File: src/core/segment_allocator.cc [hunk 1]
```diff
@@ -12,8 +12,8 @@ constexpr size_t kSegmentShift = MI_SEGMENT_SHIFT;
 namespace dfly {
 
 SegmentAllocator::SegmentAllocator(mi_heap_t* heap) : heap_(heap) {
-  // mimalloc uses 4MiB segments and we might need change this code if it changes.
-  constexpr size_t kSegLogSpan = 32 - kSegmentIdBits + 3;
+  // mimalloc uses 32MiB segments and we might need change this code if it changes.
+  constexpr size_t kSegLogSpan = 32 - kSegmentIdBits + 3 + 3;
```
Comment (kostasrim) [inline src/core/segment_allocator.cc:16]: This should be right based on the mapping in `mimalloc/types.h`.

If I missed anything plz let me know

Comment (adiholden) [reply src/core/segment_allocator.cc:16]: I believe this logic for segment allocator is broken now.
Anyway can you please find how to go to 4Mib segments again 
https://github.com/microsoft/mimalloc/pull/753

Comment (romange) [reply src/core/segment_allocator.cc:16]: @adiholden  why do you think it's broken? Basically it's about how many bits we allocate for a segment id and the segment offset. at the end they should sum up to 32.

File: src/core/segment_allocator.cc [hunk 2]
```diff
@@ -12,7 +12,7 @@ constexpr size_t kSegmentShift = MI_SEGMENT_SHIFT;
 namespace dfly {
 
 SegmentAllocator::SegmentAllocator(mi_heap_t* heap) : heap_(heap) {
-  // mimalloc uses 4MiB segments and we might need change this code if it changes.
+  // mimalloc uses 5MiB segments and we might need change this code if it changes.
```
Comment (adiholden) [inline src/core/segment_allocator.cc:15]: 4MiB :)

**Review by adiholden — APPROVED** [review]

**Pull request merged**